baseballAPI.UnitTests/Controllers/BattingControllerTests.cs

```diff
@@ -86,12 +88,23 @@ public void GetBattingStatsByYearReturnsEnumerableLeagueBattingStats()
 
             _battingService.Setup(mockPlayerService => mockPlayerService.GetBattingStatsByYear(firstPerson.YearId)).Returns(expectedRecord);
 
-            var actualReturn = _controller.GetBattingStats(firstPerson.PlayerId);
+            var actualReturn = _controller.GetBattingStatsByYear(firstPerson.YearId);
 
             Assert.That(actualReturn.ElementAt(0), Is.EqualTo(firstPerson));
             Assert.That(actualReturn.ElementAt(1), Is.EqualTo(secondPerson));
         }
 
+        [Test]
+        public void IfGetBattingByYearFailsToFindEntryItThrowsNotFoundException()
+        {
+            int badId = 1;
+            _battingService.Setup(mockPlayerService => mockPlayerService.GetBattingStatsByYear(badId)).Returns((IEnumerable<BattingLeaderBoardStats>)null);
```

Comment:
I think the real service will return an empty list if it doesn't find any matches.  Not sure if we will ever hit this case.  (I might be missing something though)

Reply:
Also may want to consider renting some variables here.  `_battingService.Setup(mockPlayerService =>`

`mockPlayerService` doesn't seem like what you want here.  (Probably copy paste miss)